### PR TITLE
fix asterisk not showing up for required string nodes, re #9546

### PR DESCRIPTION
**Pull request by apeters**
re #9546
re #9281

### CODE REVIEW

File: arches/app/templates/views/components/widgets/rich-text.htm [hunk 1]
```diff
@@ -8,7 +8,12 @@
         <div class="widget-inline-tools-collapser" data-bind=" click: function() {
             showi18nOptions(!showi18nOptions());
         }">
-            <label class="control-label widget-input-label" for="" data-bind="text:label"></label>
+            <span">
```
Comment (chiatt) [inline arches/app/templates/views/components/widgets/rich-text.htm:11]: Looks like there is an accidental quote that ended up in this tag.

File: arches/app/templates/views/components/widgets/rich-text.htm [hunk 2]
```diff
@@ -8,7 +8,12 @@
...
+                <label class="control-label widget-input-label data-bind="text:label"></label>
```
Comment (chiatt) [inline arches/app/templates/views/components/widgets/rich-text.htm:12]: A double quote is needed after `widget-input-label`

**Review by chiatt — CHANGES_REQUESTED** [review]
It looks like there are a couple of typos preventing webpack from building. Also, this fixes the UI, but the node requirement is actually not enforced if you try to save a string without a value. This is because we no longer assign `None` to a string node. Instead we save the object with an empty string for each language. It's not ideal, but it allowed us to overcome dirty state issues. 

As a result this evaluation returns False and the missing value is never identified: https://github.com/archesproject/arches/blob/b32545022825912f1651fd46f32de7da26459260/arches/app/models/tile.py#L304

One solution might be to add a datatype method that reports whether a value is empty based on the nature of the data. For the string datatype, that might mean that all values in all languages are an empty string. I think this would also make sense for other datatypes that store JSON. For example, the geojson datatype might return empty if there is a geojson object that has no features.

**Review by chiatt — APPROVED** [review]

**Comment by apeters** [comment]
@bferguso (@chiatt ), you're right it looks like the text widget is the only one where we "lost" the asterisk (the rich-text widget seems to be ok) .  Looks like it happened somewhere between dev/7.3 and dev/7.4  This may have happened as a result of the accessibility work.
This should be fixed.  I'll make a new ticket

**Pull request merged**